bccsp/factory/pkcs11.go

```diff
@@ -70,7 +70,7 @@ func setFactories(config *FactoryOpts) error {
 	}
 
 	// PKCS11-Based BCCSP
-	if config.Pkcs11Opts != nil {
+	if config.ProviderName == "PKCS11" && config.Pkcs11Opts != nil {
```

Comment:
The pkcs11 side should be made consistent with the nopkcs11 side. The `factoriesInitError` package state is not handled the same in the two methods. This is confusing and is causing odd constructs in the tests.

`factoriesInitError` should be set as a result of the once and not here.